privx_api/response.py

```diff
@@ -80,7 +80,7 @@ def __init__(
         self.status = response.status
 
         if not stream:
-            self._content = response.read().decode()
+            self._content = self._format_content()
```

Comment:
As stated earlier there are many problems with doing many things in the constructor. I recommend: 

- Do lazy evaluation of content. Only parse the return status.
- Create a unified data access by forcing the caller to do iter_content() to access stream data, even if it is not chunk-downloaded. 
- Keep raw data in byte format and delegate the processing to the caller
- Do not change the type of resp._content from byte to dict (expand the line below, L88)
- in Python 3 decode() is automatically performed with encoding = "utf-8" unless otherwise specified. But it is not needed here.Only the consumer (of PrivXStreamResponse) knows the nature of the byte stream and hence it must make sense of the data, not the PrivXStreamResponse.

```
class PrivXStreamResponse(BaseReponse):
     def __init__(self, ...):
        self.status = resp.status == expected_status
        self._response = resp
        self.stream = stream

    def iter_content(self, chunk_size=1024*1024):
        // no change, Need to handle open/closed http channel here
        pass
        
    def data(self):
       raise NotImplementedError("Should not access all data in a stream response")


# Possible usage
stream = api.download_stuff()
with open(filename, "wb") as f:
    for content in stream.iter_content():
        f.write(content)

# If caller is confident about the stream nature:
stream = api.download_stuff()
raw = b"".join([content for content in stream.iter_content(1024)])
text = raw.decode() // not PrivXStreamResponse responsibility.
```